www/js/splash/notifScheduler.js

```diff
@@ -79,6 +79,55 @@ angular.module('emission.splash.notifscheduler',
         });
     }
 
+    //new method to fetch notifications
+    scheduler.getScheduledNotifs = function() {
+        return new Promise((resolve, reject) => {
+            /* if the notifications are still in active scheduling it causes problems
+            anywhere from 0-n of the scheduled notifs are displayed 
+            if actively scheduling, wait for the scheduledPromise to resolve before fetching prevents such errors
+            */
+            if(isScheduling) 
+            {
+                console.log("requesting fetch while still actively scheduling, waiting on scheduledPromise");
+                scheduledPromise.then(() => {
+                    getNotifs().then((notifs) => {
+                        console.log("done scheduling notifs", notifs);
+                        resolve(notifs);
+                    })
+                })
+            }
+            else{
+                getNotifs().then((notifs) => {
+                    resolve(notifs);
+                })
+            }
+        })
+    }
+
+    //get scheduled notifications from cordova plugin and format them
+    const getNotifs = function() {
+        return new Promise((resolve, reject) => {
+            cordova.plugins.notification.local.getScheduled((notifs) => {
+                if (!notifs?.length){
+                    console.log("there are no notifications");
+                    resolve([]); //if none, return empty array
+                }
+                
+                const notifSubset = notifs.slice(0, 5); //prevent near-infinite listing
```

Comment:
I realized that some opcodes might have more notifications than we want to show at once, so this caps it at 5 to be displayed!

Reply:
Yes, for the year-long studies, there are definitely more notifications than we want to show at once!
You can resolve this after you have read it.